examples/rust/Sum_over_array.rs

```diff
@@ -1,3 +1,3 @@
 pub fn sum_array(x: &[i32]) -> i32 {
-  x.iter().fold(0, |sum, next| sum + *next)
```

Comment:
I think this should not be changed, so we can show that the compiler is clever enough to optimize this. What are your thoughts on this=

Reply:
Ah, I didn't realise that was the intent. Nevermind then.

Reply:
I don't think this should be closed just yet

Reply:
@CryZe is right. I think this is a good oportunity to improve Rust's examples